druid/src/widget/click.rs

```diff
@@ -52,12 +52,14 @@ impl<T: Data> Click<T> {
 impl<T: Data, W: Widget<T>> Controller<T, W> for Click<T> {
     fn event(&mut self, child: &mut W, ctx: &mut EventCtx, event: &Event, data: &mut T, env: &Env) {
         match event {
-            Event::MouseDown(_) => {
-                ctx.set_active(true);
-                ctx.request_paint();
+            Event::MouseDown(mouse_event) => {
+                if mouse_event.button == MouseButton::Left {
+                    ctx.set_active(true);
+                    ctx.request_paint();
+                }
             }
-            Event::MouseUp(_) => {
-                if ctx.is_active() {
```

Comment:
The `ctx.is_active()` check must still be there. Otherwise, you can press left click, drag the cursor off the button, release and it will be considered a click on the button. You can try this in the `calc` example.